### ISSUE TITLE
add auto-scroll

### ISSUE THREAD

**Issue by KeronCyst** [sumatrapdfreader/sumatrapdf#1680]
I came here hoping that Sumatra PDF would have a more fine-tunable auto-scroll feature than Adobe Acrobat but it doesn't seem to have one at all.

PDFs can be gradually auto-scrolled through in Acrobat via `Ctrl+Shift+H`, with the arrow keys being used to adjust scrolling speed. I'm hoping that Sumatra can have a similar auto-scroll feature with an even more finely adjustable scrolling speed.

**Comment by GitHubRulesOK** [comment]
You are correct there is no keyboard auto scroll only the middle mouse button one.

PDF readers like Acrobat do not work well with autoscroll, for a simple reason that they render glyphs line by line and step down a whole line, rather than a game or image viewer that renders pixels line by pixel line.

There are many complaints that when the mouse or tablet are autoscrolling it is "jerky" or "jittery" this could possibly be avoided by using large buffers to pre-empt a  page loading then convert the screen full of data as a fixed width set of images like a comic book then scroll those pixel by pixel. 
However that is not the way Windows sends auto scroll commands, so even comic books jump several ines at a time.
Try autoscroll in MSPaint and note that for me it jumps 10 lines at a time !!

**Comment by GitHubRulesOK** [comment]
@kjk 
One way I think this might be achieved is some means to drive some "AutoGoTo" = Previous page Y+dy where dy is a user choice to control speed, at worst there would be a pause between pages ?

**Comment by SumatraPeter** [comment]
I'm pretty sure we have a plethora of issues already that we can close this as a dupe of, such as #248 #606 #672 #873 #1083 #1498 or perhaps #167 #557 #1062...

**Comment by GitHubRulesOK** [comment]
@SumatraPeter 
I let it ride open as #557 or #606 which you mention, are related to mouse/touchpad (which I indicated as supported) 
However this is a request for key-in similar to the one in Acrobat which uses a different method to the hardware ones.

**Comment by GitHubRulesOK** [comment]
@kjk the CmdStartAutoScroll in
SumatraPDF AutoScroll "Works Well" for me (having used pointers and mouse or puck in hundreds of systems ! not just desktops) 

But then again for personal use I tend to choose a mouse (cheap does not mean poor) that has a fine level of control "on the mouse" so the arrow keys are best kept for line scrolling as faster or move the mouse more to alter the dynamic speed of scroll.

**Comment by kjk** [comment]
We already have this via the command palette (3.7+).

`Ctrl + K`, type `auto scroll`, pick **Start Auto-Scroll** (`CmdStartAutoScroll`). That starts middle-click-style auto-scroll at the cursor — move the mouse away from that point to scroll (farther = faster). Run it again (or middle-click) to stop.

To bind it to `Ctrl + Shift + H` like Acrobat, use Settings / Advanced Options… and add:

```
Shortcuts [
    [
        Cmd = CmdStartAutoScroll
        Key = Ctrl + Shift + H
    ]
]
```
